mark.c

```diff
@@ -1651,15 +1683,45 @@ GC_API void GC_CALL GC_push_all_eager(void *bottom, void *top)
     if (top == 0) return;
 
     /* Check all pointers in range and push if they appear to be valid. */
-      lim = t - 1 /* longword */;
-      for (p = b; (word)p <= (word)lim;
-           p = (word *)(((ptr_t)p) + ALIGNMENT)) {
-        REGISTER word q = *p;
+    lim = t - 1 /* longword */;
+    for (p = b; (word)p <= (word)lim;
+         p = (word *)(((ptr_t)p) + ALIGNMENT)) {
+      REGISTER word q = *p;
 
-        GC_PUSH_ONE_STACK(q, p);
-      }
+      GC_PUSH_ONE_STACK(q, p);
+    }
+
+#   undef GC_greatest_plausible_heap_addr
+#   undef GC_least_plausible_heap_addr
+# else 
```

Comment:
I struggled to follow the `#ifdef` chain: perhaps explicitly saying `elif defined(__CHERI_PURE_CAPABILITY)` here would help?

Reply:
Annotated the `# ifdef`s with comments in commit [3d1414c](https://github.com/capablevms/bdwgc/pull/13/commits/3d1414c66d1106e3a54a10e2c4e3abcc8d426fbe)